pkg/app/piped/planner/planner.go

```diff
@@ -37,8 +37,10 @@ type Planner interface {
 }
 
 type Input struct {
-	// Readonly deployment model.
-	Deployment                     *model.Deployment
+	ApplicationID                  string
+	ApplicationName                string
+	GitPath                        model.ApplicationGitPath
+	Trigger                        model.DeploymentTrigger
```

Comment:
Passing only needed data to make it can be reused by `planpreview` package where there is no deployment data.